doc/release/1.12.0-notes.rst

```diff
@@ -97,3 +97,9 @@ Changes
 
 Deprecations
 ============
+
+`np.iterable` has been deprecated.
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+This is a very old function that has some limited internal use but
+should not be a part of the public API. It can still be accessed
+internally as `numpy.function_base._iterable`.
```

Comment:
release notes are for public consumption, so shouldn't mention the internal API unless we want people to use it (which we don't)